packages/asset-swapper/src/utils/market_operation_utils/bancor_service.ts

```diff
@@ -0,0 +1,36 @@
+import { BigNumber } from '@0x/utils';
+import { SDK } from '@bancor/sdk';
+import { BlockchainType, Token } from '@bancor/sdk/dist/types';
+
+import { BancorQuoteData } from './types';
+
+function token(address: string, blockchainType: BlockchainType = BlockchainType.Ethereum): Token {
+    return {
+        blockchainType,
+        blockchainId: address,
+    };
+}
+
+export class BancorService {
+    // Bancor recommends setting this value to 2% under the expected return amount
+    public minReturnAmountBufferPercentage = 0.98;
+    private _sdk?: SDK;
+
+    constructor(public ethereumNodeEndpoint: string) {}
+
+    public async getSDKAsync(): Promise<SDK> {
+        if (!this._sdk) {
+            this._sdk = await SDK.create({ ethereumNodeEndpoint: this.ethereumNodeEndpoint });
+        }
+        return this._sdk;
+    }
+
+    public async getQuoteAsync(fromToken: string, toToken: string, amount: BigNumber): Promise<BancorQuoteData> {
+        const sdk = await this.getSDKAsync();
+        const { path, rate } = await sdk.pricing.getPathAndRate(token(fromToken), token(toToken), amount.toString());
```

Comment:
does this make an RPC request for every amount? That seems like a lot of (13) requests?

Reply:
I *think* it actually caches the reserve graph and finds paths based off that. https://github.com/bancorprotocol/sdk/blob/453a13bf6d7d61e7136a004eabf6bc1f242b6f0e/src/blockchains/ethereum/index.ts#L77

This does seem to imply that we need a timer to routinely call `sdk.refresh()` though.

Reply:
yeah it does :/ paths are cached but each call to `getPathAndRate` refreshes the available rates in liquidity pools. basically every call to `getPathAndRate` results in 1 RPC call.

Reply:
chatted off-thread. we are going to leave it this way for now, pending bancor SDK upgrade to accept multiple amounts.